geotiepoints/simple_modis_interpolator.py

```diff
@@ -0,0 +1,251 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Interpolate MODIS 1km navigation arrays to 250m and 500m resolutions.
+
+The code used here is a rewrite of the IDL function ``MODIS_GEO_INTERP_250``
+used by Liam Gumley. It has been modified to convert coordinates to cartesian
+(X, Y, Z) coordinates first to avoid problems with the anti-meridian and poles.
+This code was originally part of the CSPP Polar2Grid project, but has been
+moved here for integration with Satpy and newer versions of Polar2Grid.
+
+This algorithm differs from the one in ``modisinterpolator`` as it only
+requires the original longitude and latitude arrays. This is useful in the
+case of reading the 250m or 500m MODIS L1b files or any MODIS L2 files without
+including the MOD03 geolocation file as there is no SensorZenith angle dataset
+in these files.
+
+"""
+
+from functools import wraps
+
+import numpy as np
+from scipy.ndimage.interpolation import map_coordinates
+from .geointerpolator import lonlat2xyz, xyz2lonlat
+
+try:
+    import dask.array as da
+except ImportError:
+    # if dask can't be imported then we aren't going to be given dask arrays
+    da = None
+
+try:
+    import xarray as xr
+except ImportError:
+    xr = None
+
+# MODIS has 10 rows of data in the array for every scan line
+ROWS_PER_SCAN = 10
+
+
+def scanline_mapblocks(func):
+    """Convert dask array inputs to appropriate map_blocks calls.
+
+    This function, applied as a decorator, will call the wrapped function
+    using dask's ``map_blocks``. It will rechunk inputs when necessary to make
+    sure that the input chunks are entire scanlines to avoid incorrect
+    interpolation.
+
+    """
+    @wraps(func)
+    def _wrapper(lon_data, lat_data, res_factor=4):
+        if lon_data.ndim != 2 or lat_data.ndim != 2:
+            raise ValueError("Expected 2D lon/lat arrays.")
+        if hasattr(lon_data, "compute"):
+            # assume it is dask or xarray with dask, ensure proper chunk size
+            # if DataArray get just the dask array
+            lon_dask = lon_data.data if hasattr(lon_data, "dims") else lon_data
+            lat_dask = lat_data.data if hasattr(lat_data, "dims") else lat_data
+            lon_dask, lat_dask = _rechunk_lonlat_if_needed(lon_dask, lat_dask)
+            new_lons, new_lats = _call_map_blocks_interp(func, lon_dask, lat_dask, res_factor)
+            if hasattr(lon_data, "dims"):
+                # recreate DataArrays
+                new_lons = xr.DataArray(new_lons, dims=lon_data.dims)
```

Comment:
How about the attributes?

Reply:
I thought I would take the xarray approach on this and lose the attributes assuming they no longer apply to the new data. Thoughts?

Reply:
I'd like it to be more concervative. With interpolation, the only thing that changes are the coordinates, but they are not included here. But I don't know

Reply:
I would think not including the attributes is the more conservative (strict/rigid) way of approaching this as it is more inline with xarray logic, but yeah definitely not conservative in the "conserve" the metadata sense.